src/constantProductCalculator.ts

```diff
@@ -101,4 +148,37 @@ export class ConstantProductCalculator implements SwapCalculator {
       "Amount of minted liquidity tokens must be greater then 0.",
     );
   }
+
+  getZapParams(zapAmount: bigint, input: ZapInput): ZapParams {
+    const totalA =
```

Comment:
There's a `PoolCalculator.getLiquidities` method that does pretty much the same. Make it public or extract it to a utility function.